icevision/models/torchvision_models/faster_rcnn/show_results.py

```diff
@@ -31,3 +36,103 @@ def show_results(
+    In case of faster_rcnn those are loss_classifier, loss_box_reg and loss_total
+    (sum of the first 2 plus RPN related losses, e.g. sum of 4 terms).
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+
+    Returns
+    -------
+    samples_plus_losses: those are the samples (dictionaries) contained in the original dataset
+                          enriched with losses.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    device = torch.device("cpu")
+    model = model.to(device)
+    dl = valid_dl(dataset, batch_size=1, num_workers=0, shuffle=False)
+
+    torch.manual_seed(0)
+    samples_plus_losses = []
+    losses_stats = {
+        "loss_classifier": [],
+        "loss_box_reg": [],
+        "loss_objectness": [],
+        "loss_rpn_box_reg": [],
+        "loss_total": [],
+    }
+    for_text = ["loss_classifier", "loss_box_reg", "loss_total"]
+
+    with torch.no_grad():
+        for (x, y), sample in pbar(dl):
+            loss = model(x, y)
+            loss = {k: float(v.numpy()) for k, v in loss.items()}
+            loss["loss_total"] = sum(loss.values())
+
+            text = ""
+            for l in losses_stats.keys():
+                losses_stats[l].append(loss[l])
+                if l in for_text:
+                    text += f"{l}: {round(loss[l], 5)}\n"
+
+            text += f"IMG: {sample[0]['filepath'].name}"
+            loss["text"] = text
+
+            sample[0].update(loss)
+            samples_plus_losses.append(sample[0])
+
+    losses_stats = {k: get_stats(v) for k, v in losses_stats.items()}
+    return samples_plus_losses, losses_stats
+
+
+def plot_top_losses(
+    model: nn.Module,
+    dataset: Dataset,
+    sort_by: str = "loss_total",
+    n_samples: int = 5,
+) -> Tuple[List[dict], List[dict], dict]:
+    """
+    Gets a dataset and a model as input. Calculates losses for each sample in the dataset.
+    Sorts samples by `sort_by` (e.g. one of the losses). Runs model predictions on samples.
+    Plots the top sorted `n_samples` and returns losses + predictions + losses_stats.
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+    sort_by: (str) the loss to sort samples by
+    n_samples: how many samples to show
+
+    Returns
+    -------
+    sorted_samples: those are the samples (dictionaries) contained in the original dataset
+                    enriched with losses and sorted by `sort_by`.
+    sorted_preds: `model` predictions on `sorted_samples`. Same order as `sorted_samples`.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    model_inf = deepcopy(
+        model
+    )  # we need to create a copy of the model as `get_preds` puts it in eval mode and we need it in training mode
```

Comment:
I have not understood yet why we need to deepcopy the model. So, `get_losses` is first called with the model in `training` mode, then `get_preds` get's called and only then the model is put into `eval` mode.

If it's necessary we can put the model in training mode again after `get_preds`

Reply:
`get_preds` put the model in `eval` mode, meaning that the forward will now calculate predictions and not losses.
So when calling `plot_top_losses` again, in the same notebook/script, without re-instantiating the model, the function will error out.

This is why I am copying the model.
I can indeed set `model.training = True` after `get_preds` and it should work without copying!

Reply:
Given the screenshot I shared above, setting `model.training = True` after `model.eval()` might not work.
I need to take a closer look but we might be screwing up the internal inner workings of torchvision.

Reply:
Done. Removed `deepcopy` and added `model.train()`.